components/mpas-ocean/src/shared/mpas_ocn_wetting_drying.F

```diff
@@ -356,40 +363,207 @@ subroutine ocn_wetting_drying_wettingVelocity(layerThickEdgeFlux, layerThickness
+
+                    wettingVelocityFactor(1,iEdge) = 1.0_RKIND
+
+                    cellDummy(1:2) = cellsOnEdge(1:2,iEdge)
+
+                    cellCur = iCell
+                    cellNei = merge( cellDummy(2), cellDummy(1), iCell == cellDummy(1) )
+                    if ( cellNei > nCellsAll ) cellNei = cellCur
+
+                    sshCur = sshCell(cellCur)
+                    sshNei = sshCell(cellNei)
+
+                    ! if the neigbor cell is anticipted wet
+                    if ( layerThicknessCell(cellNei) >  &
+                            hCrit + config_drying_safety_height ) then
+                      if ( sshCur < sshNei ) then
+                         wettingVelocityFactor(1,iEdge) = 0.0_RKIND
+                      end if
+                    end if
+
+                 end if
+               end do !iEdge
+
+            end if !ifDryCell
+        end do
+
+        ! Ramping
+        hRampMin = config_zero_drying_velocity_ramp_hmin
+        do iCell = 1, nCellsAll
+
+            if ( config_zero_drying_velocity_ramp .and. &
+                  (layerThicknessCell(iCell)  > &
+                   hCrit + config_drying_safety_height) .and. &
+                  (layerThicknessCell(iCell) <= config_zero_drying_velocity_ramp_hmax) ) then
+
+                 ! Following O'Dea et al. (2021), if total upwinded wct is less than
+                 ! 2*critical thickness, apply damping at each edge
+                 do i = 1, nEdgesOnCell(iCell)
+                    iEdge = edgesOnCell(i, iCell)
+                    if ( normalVelocity(1, iEdge) * edgeSignOnCell(i, iCell) <= 0.0_RKIND ) then
+                      wettingVelocityFactor(1, iEdge) = 1.0_RKIND - &
+                          tanh(50.0_RKIND * (layerThicknessCell(iCell) - hRampMin)/hRampMin)
+                    end if
+
+                    ! If edge velocity has been zeroed, check neighbor and current cell ssh.
+                    ! Prevent flow out of cell if neighbor cell is dry and the ssh gradient
+                    ! would cause flow from neighbor cell into current cell.
+                    if ( normalVelocity(1, iEdge) * edgeSignOnCell(i,iCell) == 0.0_RKIND ) then
+                      cellDummy(1:2) = cellsOnEdge(1:2,iEdge)
+
+                      cellCur = iCell ;
+                      cellNei = merge( cellDummy(2), cellDummy(1), iCell == cellDummy(1) )
+                      if ( cellNei > nCellsAll ) cellNei = cellCur
+
+                      sshCur = sshCell(cellCur)
+                      sshNei = sshCell(cellNei)
+
+                      ! if the neigbor cell is anticipated dry
+                      if ( layerThicknessCell(cellNei) <=  &
+                          hCrit + config_drying_safety_height ) then
+                         if ( sshCur < sshNei ) then
+                             wettingVelocityFactor(1,iEdge) = 1.0_RKIND
+                         end if
+                      end if
+                    end if
```

Comment:
I might be tracking the if-then clause bounds poorly online, but does this mean that the ssh gradient option is always on with the ramp features? We might want an additional if-clause here to allow them to be turned on separately

Reply:
I believe the ramp and ssh gradient option should be independent as-is. The ramp feature is still optional inside the `config_use_modify_wetting_drying_implementation`/`config_use_ssh_gradient_wetting_drying` section since there is an `if ( config_zero_drying_velocity_ramp ...` around the ramping factor code. In the ssh gradient option + ramp feature case, there is an additional check to make sure the ramping factor respects the need to limit flow out of a dry neighbor cell based despite an ssh gradient that would otherwise cause this to occur (for a previously dry edge). Let me know if I'm missing something though!

Reply:
Oh, I see. I forgot that the ramp was applied in `ocn_wetting_velocity_factor_on_cell_edges`. Thanks!

Reply:
I was also confused about that initially, since that routine was added after I had last looked carefully at the W/D in master.